src/main/java/io/stargate/sgv2/jsonapi/api/model/command/deserializers/FilterClauseDeserializer.java

```diff
@@ -111,6 +111,15 @@ private void populateExpression(
       }
       logicalExpression.addLogicalExpression(innerLogicalExpression);
     } else {
+      // the key should match pattern or equal to $vector
+      if (!(DocumentConstants.Fields.VALID_PATH_PATTERN.matcher(entry.getKey()).matches()
+          || entry.getKey().equals("$vector"))) {
+        throw new JsonApiException(
+            ErrorCode.INVALID_FILTER_EXPRESSION,
+            String.format(
+                "%s: filter clause path ('%s') contains character(s) not allowed",
+                ErrorCode.INVALID_FILTER_EXPRESSION.getMessage(), entry.getKey()));
+      }
```

Comment:
@maheshrajamani The `else` block here is for `$eq` missing, like `{ "find": { "filter" : {"name" : "1"} }}`. Should I allow `$vector` here? If it is, the case `{ "find": { "filter" : {"$vector" :  true}} }` which is not valid but will return empty data `{
    "data": {
        "documents": [],
        "nextPageState": null
    }
}`

Reply:
No, only `$exists` supported for `$vector` in filter

Reply:
Solved, could you help me review it again? Thank you!